### DISCUSSION TITLE
posts/hetgnn/

### DISCUSSION THREAD

**Discussion by giscus** [ZZy979/zzy979.github.io#2] [Comments]
# posts/hetgnn/

Heterogeneous Graph Neural Network

https://zzy979.github.io/posts/hetgnn/

**Comment by dd-python** [comment]
作者您好，我想问一下你关于这篇论文的自己的代码实现。您编写的代码里面train部分中的训练方式和原文作者给的好像有不同。原文中是采用三元组的方式进行训练，而您的代码是用连边来进行训练，请问这两种方法有什么不同吗？

**Reply by ZZy979** [reply]
论文中的训练方式是对每个顶点v，通过随机游走得到邻居vc，再通过负采样得到vc'，组成公式(11)中的三元组<v, vc, vc'>，然后使用交叉熵损失进行训练。

我的代码中，首先用random_walk.py通过随机游走采样邻居vc，构造(v, vc)组成的邻居图。然后在train.py中对这个邻居图通过负采样得到vc'。所以和论文中描述的训练方式是一致的。
